### ISSUE TITLE
NixOS: Add documentation on recovering root ZFS with instalation media

### ISSUE THREAD

**Issue by mariaa144** [openzfs/openzfs-docs#414]
Provide a section which explains how to use a NixOS USB to boot and correctly mount a root ZFS setup.

I have had a couple of instances where I needed to boot my setup system but I didn't know how to properly mount ZFS and rebuild the environment. The first time I didn't know my root password on the first install. The second time I didn't have networking working on the first install so I couldn't rebuild with the right packages.

**Comment by ne9z** [comment]
I have added instructions on host name here:

https://github.com/ne9z/dotfiles-flake/blob/0dacedd7c3a28f1b3ac12c710767c445432867c1/hosts/exampleHost/changeHostName.txt

> Provide a section which explains how to use a NixOS USB to boot and
> correctly mount a root ZFS setup.  I have had a couple of instances
> where I needed to boot my setup system but I didn't know how to
> properly mount ZFS and rebuild the environment.

This is not needed, as novice users can just select a earlier working
generation at the GRUB menu, provided that the first installation is
working.

Also, this information essentially identical to the part after disk
partitioning.  Advanced users should be able to mount the datasets
referring to the instructions there.

> The first time I didn't know my root password on the first install.

This password is set by yourself during installation.

> The second time I didn't have networking working on the first install
> so I couldn't rebuild with the right packages.

I recommend using NetworkManager, if you are unfamiliar with configuring
network with wpa-supplicant or dhclient.

**Comment by mariaa144** [comment]
I agree it isn't needed but it would be helpful I think. In my case I had no earlier snapshots because it was original install. Perhaps I'm not an advanced user. I ran into trouble trying to figure out how to mount all the partitions correctly so I could rebuild.

For the password, I think I had a special character in my password which broke the script and made my password different than I expected. It happened on an older version of the guide which doesn't have the same issue now.

I'm only using a wire and DNS wasn't working by default without NetworkManager. This surprised me. I also tried to enable `systemd-resolved.service` but still no DNS. Reinstalled with NetworkManager enable worked for me. I didn't try `dhclient`.
